### ISSUE TITLE
Add data to the API

### ISSUE THREAD

**Issue by msaraiva** [msaraiva/surface#27]
Similar to `property`, I want to add `data` and `context` functions so we can benefit from static checking and better tooling. Proposal:

### Data (same concept of Vue.js)

```elixir
data show, :boolean, default: false
```

The `:default` option could minimise the need to implement `mount/3`.

### Context (same existing concept, but more restrict and explicit)

```elixir
defmodule Form do
  ...
  context set: :form

end
```

```elixir
defmodule TextInput do
  ...
  context get: :form, from: Form

end
```

Both `data` and `context` would be merged into the assigns. So instead of `@context.form` we could use `@form` directly.

We can also have a `:as` option for cases when we want to change the name of the assign:
 
```elixir
defmodule TextInput do
  ...
  context get: :form, from: Form, as: :whatever

end
```

**Comment by msaraiva** [comment]
The `data` part has been addressed in #28

**Comment by msaraiva** [comment]
I'm closing this one and opening a new separate issue to discuss the changes in the `context` API.
